zap/src/main/dist/zap.sh

```diff
@@ -22,7 +22,9 @@ OS=$(uname -s)
 # If we're on OS X, try to use the bundled Java; if it's not there, then the system Java
 # Life would be much easier if OS X had readlink -f
 if [ "$OS" = "Darwin" ]; then
-  if [ -e ../PlugIns/jre*/Contents/Home/bin/java ]; then
+  OSX_VERSION=$(sw_vers -productVersion)
+  OSX_MAJOR_VERSION=${OSX_VERSION%%[.|-]*}
+  if [ -e ../PlugIns/jre*/Contents/Home/bin/java && $OSX_MAJOR_VERSION -lt 13]; then
```

Comment:
Based on latest comments from the issue, the correct fix in the script is to check here if we are able to use this Java instead of skipping based on macOS version.